src/Turing.jl

```diff
@@ -30,8 +30,8 @@ include("stdlib/distributions.jl")
 include("stdlib/RandomMeasures.jl")
 include("utilities/Utilities.jl")
 using .Utilities
-include("core/Core.jl")
-using .Core
+include("essential/Essential.jl")
+using .Essential
```

Comment:
```suggestion
include("essential/Essential.jl")
Base.@deprecate_binding Core Essential
using .Essential
```

Reply:
Actually, it does _export_ `Code` :confused: 

```julia
julia> @macroexpand Base.@deprecate_binding Core Essential
:($(Expr(:toplevel, :(export Core), :(const _dep_message_Core = ", use Essential instead."), :(const Core = Essential), :(Base.deprecate(Main, :Core)))))
```

Reply:
We shouldn't export it though:
```suggestion
include("essential/Essential.jl")
Base.@deprecate_binding Core Essential false
using .Essential
```

Reply:
```julia
julia> @macroexpand Base.@deprecate_binding Core Essential false
:($(Expr(:toplevel, nothing, :(const _dep_message_Core = ", use Essential instead."), :(const Core = Essential), :(Base.deprecate(Main, :Core)))))
```

Reply:
Haha, yeah I just realized :sweat_smile: